src/backend/distributed/planner/multi_explain.c

```diff
@@ -421,6 +462,32 @@ RemoteExplain(Task *task, ExplainState *es)
 
 	RemoteExplainPlan *remotePlan = (RemoteExplainPlan *) palloc0(
 		sizeof(RemoteExplainPlan));
+
+	/*
+	 * For EXPLAIN EXECUTE we still use the old method, so task->fetchedExplainAnalyzePlan
+	 * can be NULL for some cases of es->analyze == true.
+	 */
+	if (es->analyze && task->fetchedExplainAnalyzePlan)
+	{
+		/*
+		 * Similar to postgres' ExplainQuery(), we split by newline only for
+		 * text format.
+		 */
+		if (es->format == EXPLAIN_FORMAT_TEXT)
+		{
+			remotePlan->explainOutputList = SplitString(task->fetchedExplainAnalyzePlan,
+														'\n');
+		}
+		else
+		{
+			remotePlan->explainOutputList = list_make1(task->fetchedExplainAnalyzePlan);
+		}
+
+		remotePlan->placementIndex = task->fetchExplainAnalyzePlacementIndex;
+
+		return remotePlan;
```

Comment:
pretty minor, but when a function has two wholly different implementations I find it nicer if it's structured as :
```c
if (useImplementation1)
{
    return Implementation1();
}
else
{
    return Implementation2();
}
```